Sources/BSON/JSON.swift

```diff
@@ -0,0 +1,138 @@
+import Foundation
+
+/// Enum representing a JSON value
```

Comment:
can we add ", used internally for modeling JSON during extJSON parsing/generation" to this docstring? To people new to the codebase, it might not be immediately obvious why we need a JSON enum in a BSON library.

Reply:
yep